File: app/src/main/java/com/example/mhmdreza_j/xproject/views/market/MarketFragment.java
package com.example.mhmdreza_j.xproject.views.market;


import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;

import com.example.mhmdreza_j.xproject.R;
import com.example.mhmdreza_j.xproject.views.base_class.BaseFragment;
import com.example.mhmdreza_j.xproject.views.main_page.MainActivity;

/**
 * A simple {@link Fragment} subclass.
 */
public class MarketFragment extends BaseFragment {


    public MarketFragment() {
        // Required empty public constructor
    }


    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        View view = inflater.inflate(R.layout.fragment_market, container, false);

        ImageView homeButton = view.findViewById(R.id.homeButton);
        homeButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                if (getActivity() == null) return;
                ((MainActivity) getActivity()).startInnerFragment(R.id.navigation_coin);
            }
        });
        
        return view;
    }

}

File: app/src/main/java/com/example/mhmdreza_j/xproject/views/market/MarketFragment.kt
package com.example.mhmdreza_j.xproject.views.market


import android.os.Bundle
import androidx.fragment.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup

import com.example.mhmdreza_j.xproject.R
import com.example.mhmdreza_j.xproject.views.base_class.BaseFragment

import com.example.mhmdreza_j.xproject.views.main_page.RANKING_POSITION
import kotlinx.android.synthetic.main.score_view.*


/**
 * A simple [Fragment] subclass.
 */
class MarketFragment : BaseFragment() {


    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        super.onCreateView(inflater, container, savedInstanceState)

        // Inflate the layout for this fragment
        return inflater.inflate(R.layout.fragment_market, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        homeButton.setOnClickListener { onBackPressed() }
    }

    override fun onBackPressed() {
        if (activity == null) return
        this.parentFragment!!.navigationView.performOnClick(RANKING_POSITION)
    }
}// Required empty public constructor
